### DISCUSSION TITLE
Specs for highlights

### DISCUSSION THREAD

**Discussion by mil-ad** [ahrm/sioyek#760] [Q&A]
I've been thinking about developing a plugin to add support for showing Supernote highlights/annotations in sioyek. If you're not familiar [Supernote](https://supernote.com/) is an eink device and the way it handles PDF annotations aligns well with sioyek, i.e. they create a separate file that includes all the annotations rather than embedding them directly into the PDF. There are some [python libraries](https://github.com/jya-dev/supernote-tool) that can read their proprietary format and export it as PNG or PDF but I'm not sure how to handle things on sioyek side. I was wondering if @ahrm could point me to what format sioyek expects for annotations? Any other pointers are much appreciated.

**Comment by ahrm** [comment]
The annotations are stored in `shared.db` which is a simple sqlite database file. You can view it using database viewer software such as this: https://sqlitebrowser.org/. The schema is described here (make sure to also read the section about coordinate spaces) https://sioyek-documentation.readthedocs.io/en/latest/scripting.html#database-files.

Feel free to ask any other questions if the documentation is not clear enough.

**Comment by mil-ad** [comment]
Thanks this is super useful. Although these annotations are mostly drawing annotations from the pen so perhaps it sits at the mupdf level.

**Reply by ahrm** [reply]
The next version of sioyek will have support for drawing annotations (it is not available in public commits yet though).
